papermill/iorw.py

```diff
@@ -255,6 +261,7 @@ def read(self, path):
     def listdir(self, path):
         return self._get_client().ls(path)
 
+    @retry.retry(RateLimitException, delay=1, backoff=2, max_delay=4)
```

Comment:
Could I convince you to add a test which raises RateLimitException to prove it retries?

Reply:
Also do we need to handle non 429 errors differently? Probably not blocking but ideally we'd capture only the 429 exceptions. Not blocking a merge but thought I'd check.

Reply:
@MSeal Will add test.
Agree we probably need to handle other errors differently (No write access/invalid path, etc)

Reply:
1. I added a test. (May need some review)
2. I created a new PR on gcsfs. https://github.com/dask/gcsfs/pull/131 this will return the specific Exception on 429 error. We can probably wait and I can make the required changes, let me know how you want to proceed.
Thanks